tests/unit/plugins/test_go.py

```diff
@@ -605,27 +608,113 @@ class Options:
             env=mock.ANY,
         )
 
-    def test_build_stage_packages_default(self):
+
+class GoPluginToolSetupTest(GoPluginBaseTest):
+    def setUp(self):
+        super().setUp()
+
         class Options:
             source = "dir"
+            go_channel = "latest/stable"
+
+        self.options = Options()
+
+    def test_snap(self):
+        plugin = go.GoPlugin("test-part", self.options, self.project)
+
+        self.assertThat(plugin.build_packages, Not(Contains("golang-go")))
+        self.assertThat(plugin.build_snaps, Contains("go/latest/stable"))
+
+    def test_build_packages(self):
+        self.options.go_channel = ""
+
+        plugin = go.GoPlugin("test-part", self.options, self.project)
+
+        self.assertThat(plugin.build_packages, Contains("golang-go"))
+        self.assertThat(plugin.build_snaps, Not(Contains("go/latest/stable")))
```

Comment:
Would it be advisable to add an extra test for a specific non-latest channel to ensure that we're able to accept arbitrary input in the `go-channel` setting? I suggest adding an extra test that sets, and checks, for `go-channel: 1.11/stable`. This will be almost identical to the `test_snap` function but with the added override on `self.options.go_channel`:

```python
    def test_alternative_channel(self):
        self.options.go_channel = "1.11/stable"

        plugin = go.GoPlugin("test-part", self.options, self.project)

        self.assertThat(plugin.build_packages, Not(Contains("golang-go")))
        self.assertThat(plugin.build_snaps, Contains("go/1.11/stable"))
```

Reply:
Thanks for the catch, the validity of the channel is checked by a different component, the same one that checks for build-snaps.

Reply:
An integration test however might be a good idea.